### PR TITLE
Fix higher order deriv with inplace

### PR DESCRIPTION
**Pull request by JiabinYang**
### PR types

Bug fixes
### PR changes

Others
### Describe

Fix Double Grad program.

### CODE REVIEW

File: paddle/phi/api/yaml/legacy_backward.yaml [hunk 1]
```diff
@@ -1823,6 +1845,16 @@
     func : sinh_grad
   inplace : (out_grad -> x_grad)
 
+- backward_api : slice_double_grad
```
Comment (veyron95) [inline paddle/phi/api/yaml/legacy_backward.yaml:1848]: slice_grad maybe should add `backward : slice_double_grad` ?

Comment (JiabinYang) [reply paddle/phi/api/yaml/legacy_backward.yaml:1848]: done

File: paddle/fluid/pybind/eager.cc [hunk 2]
```diff
@@ -98,10 +98,11 @@ void EmptyTensorInitializer(TensorObject* self,
   }
 
   if (!autograd_meta->GetMutableGradNode()) {
-    VLOG(3) << "Tensor(" << name
-            << ") have not GradNode, add GradNodeAccumulation for it.";
     autograd_meta->SetGradNode(
         std::make_shared<egr::GradNodeAccumulation>(autograd_meta));
+    VLOG(3) << "Tensor(" << name
+            << ") have not GradNode, add GradNodeAccumulation"
+            << autograd_meta->GradNode() << " for it.";
```
Comment (veyron95) [inline paddle/fluid/pybind/eager.cc:105]: 如果 stop_gradient = True, 也同样创建 GradNodeAccumulation 吗？

**Review by veyron95 — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**